File: Problem_7/problem7.c
#include <stdio.h>


int main() {
    unsigned int primes[10001];
    unsigned int i = 3;
    unsigned int counter = 1;
    unsigned int isPrime;

    primes[0] = 2;

    while (counter < 10001) {
        isPrime = 1;
        for (int prime=0; prime < counter; prime++) {
            if (i%primes[prime] == 0) {
                i += 2;
                isPrime = 0;
                break;
            }
        }
        if (isPrime) {
            primes[counter] = i;
            counter += 1;
        }
    }
    printf("%d\n", primes[counter-1]);
    return 0;
}

File: 01-09/Problem_7/problem7.rs

fn main() {
    let mut primes = vec![2];
    let mut i = 3;
    let mut counter = 1;

    while counter < 10_001 {
        let mut is_prime = true;
        for prime in &primes {
            if i%prime == 0 {
                i += 2;
                is_prime = false;
                break;
            }
        }
        if is_prime {
            primes.push(i);
            counter += 1;
        }
    }
    println!("{}", primes[counter-1])
}